File: mvvm-viewbinding/src/main/java/com/everdeng/android/lib/mvvm/viewbinding/demo/MvvmActivity.java
package com.everdeng.android.lib.mvvm.viewbinding.demo;

import android.os.Bundle;
import android.view.LayoutInflater;

import androidx.annotation.Nullable;

import com.everdeng.android.lib.mvvm.viewbinding.base.BaseMvvmActivity;
import com.everdeng.android.lib.mvvm.viewbinding.databinding.ActivityMvvmBinding;

/**
 * @author allever
 */
public class MvvmActivity extends BaseMvvmActivity<ActivityMvvmBinding, MvvmViewModel> {

    @Override
    protected ActivityMvvmBinding bindView(LayoutInflater inflater) {
        return ActivityMvvmBinding.inflate(inflater);
    }


    @Override
    protected void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        mViewModel.login();
    }

    @Override
    protected void observeData() {
        mViewModel.nickName.observe(this, s -> {
            mBinding.tvNickName.setText(s);
        });
    }
}

File: mvvm/src/main/java/com/allever/android/lib/mvvm/MvvmActivity.kt
package com.allever.android.lib.mvvm

import com.allever.android.lib.mvvm.base.BaseMvvmActivity
import com.allever.android.lib.mvvm.databinding.ActivityMvvmBinding

class MvvmActivity : BaseMvvmActivity<MvvmViewModel, ActivityMvvmBinding>(){
    override fun getLayoutId(): Int = R.layout.activity_mvvm

    override fun getBindingVariable(): Int = BR.mvvmViewModel

    override fun initView() {

    }
}